### ISSUE TITLE
No notification on Exited container.

### ISSUE THREAD

**Issue by hugalafutro** [RafhaanShah/Container-Mon#2]
Hi,
I was wondering if there is a way to notify on Exited containers in same way as is already possible for Stopped?

I have large-ish docker-compose vpn media server stack (https://paste.o5.ddns.net/?01c203401ee4a8ff#47Sw1AmHptxYVkFnc5Cs41B59QRRD5R5aDdzWN6jtwg3) with lots of container dependencies. After pc reboot 70% of the time it works every time, the remaining 30% a random number of the containers doesn't start and remains in Exited state.

Would love a way to get notified if this happens. Thanks for any insights.

**Comment by RafhaanShah** [comment]
Hello!

Do you have `CONTAINERMON_CHECK_STOPPED` flag set to `true`?

As far as I can tell `exited` counts as being `stopped`, so it should be picked up!

**Comment by hugalafutro** [comment]
Yep I do, and I get notified when Stopped, but not when Exited. I think Exited results when the docker stack starts shutting down and reaches some timeout before all the containers gracefully stop and some containers get killed.

I can't reboot the pc now as folks are streaming, but next time it happens I'll try to see if there's anything in container-mon log.

**Comment by RafhaanShah** [comment]
That's really weird, I'm not sure what the difference in `exited` is supposed to be

I just tried bringing up a random container, then did `docker stop <container_name>` and checked it's status with `docker inspect <container_name>` which shows me: 

```
"State": {
            "Status": "exited",
            "Running": false,
```

and `docker ps -a`: 
```
CONTAINER ID   IMAGE           COMMAND       CREATED          STATUS 
10989fd32808   ubuntu:latest   "/bin/bash"   14 seconds ago   Exited (0) 4 seconds ago
```

and `container_mon` notified me for that

Next time it happens could you send me the output for `docker ps -a` and the logs for `docker inspect <container_name>` for one of the exited ones you didn't get a notification for?

**Comment by hugalafutro** [comment]
Will do, it might take some time, but the issue is not pressing. Only thing I can remember is It wasn't (0) after the Exited, I think it was (137) and I found this: https://betterprogramming.pub/understanding-docker-container-exit-codes-5ee79a1d58f6 which claims 137 "Indicates failure as container received SIGKILL (Manual intervention or ‘oom-killer’ [OUT-OF-MEMORY])" which sounds to me like it could be the kill when rebooting?

**Comment by RafhaanShah** [comment]
Hmm I tried it with `docker kill` too, same thing with status ` Exited (137)` 🤷‍♂️

**Comment by hugalafutro** [comment]
Oh boy do I feel stupid now. TL;DR Error was in my config

I duplicated the stack on a different vm, kept rebooting it until I got into Exited (the number was 128 though) ... and got notification within 3 minutes. It turns out I was under impression I use same settings for container-mon in all vm's, but for reasons I forgot just the one with the "issue" had checking stopped disabled.

Sorry for wasting your time!
